Framework/DataHandling/src/LoadEmptyInstrument.cpp

```diff
@@ -146,19 +194,74 @@ void LoadEmptyInstrument::exec() {
   }
 }
 
-/// Run the Child Algorithm LoadInstrument (or LoadInstrumentFromRaw)
+// Call LoadInstrument as a child algorithm
 API::MatrixWorkspace_sptr LoadEmptyInstrument::runLoadInstrument(const std::string &filename,
                                                                  const std::string &instrumentname) {
+  auto ws = WorkspaceFactory::Instance().create("Workspace2D", 1, 2, 1);
+
   auto loadInst = createChildAlgorithm("LoadInstrument", 0, 0.5);
   loadInst->setPropertyValue("Filename", filename);
   loadInst->setPropertyValue("InstrumentName", instrumentname);
   loadInst->setProperty("RewriteSpectraMap", OptionalBool(true));
-  auto ws = WorkspaceFactory::Instance().create("Workspace2D", 1, 2, 1);
   loadInst->setProperty<MatrixWorkspace_sptr>("Workspace", ws);
 
   loadInst->execute();
 
   return ws;
 }
 
+// Call LoadIDFFromNexus as a child algorithm
+API::MatrixWorkspace_sptr LoadEmptyInstrument::runLoadIDFFromNexus(const std::string &filename) {
+  const std::string instrumentEntryName{"/instrument/instrument_xml"};
+  // There are two possibilities for the parent of the instrument IDF entry
+  const std::string instrumentParentEntryName_1{"mantid_workspace_1"};
+  const std::string instrumentParentEntryName_2{"raw_data_1"};
+  std::string instrumentParentEntryName{instrumentParentEntryName_1};
+
+  // Test if instrument XML definition exists in the file
+  bool foundIDF{false};
+  try {
+    ::NeXus::File nxsfile(filename);
```

Comment:
Unless there is a local naming conflict, I don't think `::` is needed before `NeXus`